File: frontend/src/types/driver.ts
export interface DriverSummary {
  driverId: number;
  driverRef: string;
  fullName: string;
  code: string | null;
  number: number | null;
  nationality: string | null;
  wins: number;
  podiums: number;
  championships: number;
  winPercentage: number;
}

export interface ResultsBreakdown {
  wins: number;
  otherPodiums: number;
  pointsFinishes: number;
  noPointsFinishes: number;
  dnfs: number;
}

export interface DriverDetail {
  summary: DriverSummary;
  totalRaces: number;
  averageFinish: number;
  dnfPercentage: number;
  winsBySeasonChart: { season: number; wins: number }[];
  resultsBreakdown: ResultsBreakdown;
}

File: backend/app/schemas/driver.py
"""
Driver response schemas.

Field names and nesting here match `frontend/src/types/driver.ts`
EXACTLY (DriverSummary, DriverDetail, ResultsBreakdown) — those types
were written in Milestone 5 against this eventual real API, so matching
them precisely is what lets the frontend's mock `getDrivers()` /
`getDriverDetail()` be replaced by real `fetch()` calls with no
reshaping logic in between.
"""

from typing import List, Optional

from app.schemas.base import CamelModel


class DriverSummary(CamelModel):
    driver_id: int
    driver_ref: str
    full_name: str
    code: Optional[str]
    number: Optional[int]
    nationality: Optional[str]
    wins: int
    podiums: int
    championships: int
    win_percentage: float


class SeasonWins(CamelModel):
    season: int
    wins: int


class ResultsBreakdown(CamelModel):
    wins: int
    other_podiums: int
    points_finishes: int
    no_points_finishes: int
    dnfs: int


class DriverDetail(CamelModel):
    summary: DriverSummary
    total_races: int
    average_finish: float
    dnf_percentage: float
    wins_by_season_chart: List[SeasonWins]
    results_breakdown: ResultsBreakdown
